### DISCUSSION TITLE
How to debug baresip with vscode+gdb on Linux

### DISCUSSION THREAD

**Discussion by gbernaldo82** [baresip/baresip#2575] [Q&A] [answered]
Hi, good afternoon!

I would like to be able to debug baresip (built with debug symbols) under Visual Studio Code. I need to be able to debug a problem I'm having with SIP_INFO between baresip and my software. So my question is about how can I debug baresip with gdb under Visual Studio Code.

I have built baresip (and re and rem) as described in the Building section in the page (https://github.com/baresip/baresip) but after starting baresip inside vscode with gdb and set a breakpoint in the main() function I start the debugging session but the vscode does not stop on the breakpoint. It just begin the execution of baresip as normal.

Can anyone guide me to debug baresip with VSCode and gdb please?

Thank you so much in advance!

**Comment by sreimers** [comment, marked as answer]
I think you can follow generic cmake howtos like this with `launch.json` (exec `build/baresip` and args like `-v` needs to be modified) and `tasks.json`:

https://gist.github.com/gabrielhamel/5bbde2878e81d99b0651a18a5b5e38de
